src/core/model/Document.cpp

```diff
@@ -168,7 +168,7 @@ auto Document::createSaveFilename(DocumentType type, const std::string& defaultS
             c = '_';
         }
     }
-    fs::path p = filename;
```

Comment:
Filename is not necessarily UTF-8.

Reply:
Since MSVC does weird things here, I would go for 2. 
Can you also file a bug report for MSVC, so they can fix that?
Can't believe, they did that intentionally.